[DM-5340] Document required roles for loriot endpoints (#4427)
SoftwareAG/c8y-docs: content/device-integration/lpwan-bundle/loriot-lora.md

The Loriot LoRa agent only processes "gw" messages to avoid duplicate measurements or events in {{< product-c8y-iot >}}, because most of the information matches with "gw" message whereas "gw" message also carries all gateway information.

{{< c8y-admon-info >}}

> You must enable the "gw" message option on Loriot Network Server while connecting to the Loriot LoRa agent, see [Device registration via {{< product-c8y-iot >}}](#device-registration-via-the-platform).

Reviewer: ```suggestion
You must enable the "gw" message option on the Loriot Network Server while connecting to the Loriot LoRa agent, see [Device registration via {{< product-c8y-iot >}}](#device-registration-via-the-platform).
```

In order to be able to send downlink operations, the devices registered via uplink message must be re-registered using {{< product-c8y-iot >}} (see [Device registration via {{< product-c8y-iot >}}](#device-registration-via-the-platform)), to be associated with a connection and a device type.

{{< c8y-admon-info >}}

> In order to send an uplink request, the user configured in the Loriot Network Server must have the Loriot admin role permission in {{< product-c8y-iot >}}.

Reviewer: ```suggestion
In order to send an uplink request, the user configured in the Loriot Network Server must have ADMIN permission for the "Loriot" permission type in {{< product-c8y-iot >}}.
```

> For details, see [Assigning the Loriot admin role permission](#assigning-the-loriot-admin-role-permission).

Reviewer: ```suggestion
For details, see [Assigning ADMIN permission for the Loriot](#assigning-the-loriot-admin-role-permission).
```

Reviewer: Must also be modified in the respective heading where the link points to

> For details, see [Assigning ADMIN permission for the Loriot](#assigning-the-loriot-admin-role-permission).

Reviewer 2: uplink request is sent from the Loriot platform with a user set by the Loriot Microservice.

Does sb. needs to do a manual intervention there?

Author: there is a section in the user doc about configuring the user credentials https://cumulocity.com/docs/device-integration/lpwan/#configuring-the-loriot-endpoint-using-basic-authentication. This is a subsection of "Device registration via uplink message".
Not sure if this is outdated?

Reviewer 2: shouldn't you then add this detail under
"Use Cumulocity data forwarder for configuring the Loriot endpoint using basic authentication."
?

Author: updated now

![Delete connection](/images/device-protocols/lora-loriot/loriot-admin-settings-delete.png)

{{< c8y-admon-info >}}
To create, update, or delete LNS connections, you must have the Loriot admin permission assigned to your user. In order to view the lns connections, the loriot read permission is sufficient.
{{< /c8y-admon-info >}}

#### Loriot device registration {#loriot-device-registration}

To register a Loriot device in {{< product-c8y-iot >}} navigate to **Devices** > **Registration** in the Device Management application, click **Register device** at the top right and select **Single device registration** > **Loriot LoRa** from the dropdown.

{{< c8y-admon-req >}}

> To register a device, you need the write or admin permission for "Inventory", or the admin permission for "Loriot".

Reviewer: ```suggestion
To register a device, you need WRITE or ADMIN permission for the "Inventory" permission type, or ADMIN permission for the "Loriot" permission type.
```

![Delete connection](/images/device-protocols/lora-loriot/loriot-admin-settings-delete.png)

{{< c8y-admon-info >}}

> To create, update, or delete LNS connections, you must have the Loriot admin permission assigned to your user. In order to view the lns connections, the loriot read permission is sufficient.

Reviewer: ```suggestion
To create, update, or delete LNS connections, you must have the Loriot ADMIN permission assigned to your user. To view the LNS connections, the Loriot READ permission is sufficient.
```

3. A prompt will appear stating that in order to migrate the device from one LNS connection to another, you must re-register the device. Click the **Re-Register** button.
4. You are directed to the device registration page where you can perform the re-registration following the steps above and selecting the desired LNS connection.

> ### Assigning ADMIN permission for the Loriot {#assigning-the-loriot-admin-role-permission}

Reviewer: ```suggestion
### Assigning ADMIN permission for Loriot {#assigning-the-loriot-admin-role-permission}
```

- **Spreading factor** - The <code>c8y_SpreadingFactor</code> fragment of the device managed object is updated to capture the spreading factor of the device.
- **Signal strength** - A measurement is created with RSSI and SNR values of the device signal strength.

> ### Required permissions for Loriot agent REST API {#loriot-agent-rest-api-permissions}

Reviewer 3: The rest of this documentation is user-docs, describing how to configure details in UI. These users don't know which of their actions perform any of these API requests. Also, I'd like to avoid documenting these endpoints publicly in general.

If i we create this kind of documentation, then it should be an addition to each section above and below detailing the required permissions for the user interaction.
